dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/ajax/CMSMaintenanceAjax.java

```diff
@@ -376,4 +428,293 @@ public void zipTempDirectoryToStream(OutputStream out) throws IOException {
                 CloseUtils.closeQuietly(out);
             }
 		}
+
+	/**
+	 * This method will pull a list of all tables /classed being managed by
+	 * hibernate and export them, one class per file to the backupTempFilePath
+	 * as valid XML. It uses XStream to write the xml out to the files.
+	 *
+	 * @throws ServletException
+	 * @throws IOException
+	 * @author Will
+	 * @throws DotDataException
+	 */
+	@SuppressWarnings("unchecked")
+	public void createXMLFiles() throws ServletException, IOException, DotDataException, DotSecurityException {
+		validateUser();
+		Logger.info(this, "Starting createXMLFiles()");
+
+		Set<Class> _tablesToDump = new HashSet<Class>();
+		try {
+
+			/* get a list of all our tables */
+			Map map = HibernateUtil.getSession().getSessionFactory().getAllClassMetadata();
+			Iterator it = map.entrySet().iterator();
+			while (it.hasNext()) {
+				Map.Entry pairs = (Map.Entry) it.next();
+				Class x = (Class) pairs.getKey();
+				if (!x.equals(Inode.class) && !x.equals(Clickstream.class) && !x.equals(ClickstreamRequest.class)
+						&& !x.equals(Plugin.class) && !x.equals(PluginProperty.class))
+					_tablesToDump.add(x);
+
+			}
+			XStream _xstream = null;
+			HibernateUtil _dh = null;
+			List _list = null;
+			File _writing = null;
+			BufferedOutputStream _bout = null;
+
+			for (Class clazz : _tablesToDump) {
+				if(clazz.equals(Structure.class) || clazz.equals(Field.class) || clazz.equals(FieldVariable.class)){
+					continue;
+				}
+				//http://jira.dotmarketing.net/browse/DOTCMS-5031
+				if(PermissionReference.class.equals(clazz)){
+					continue;
+				}
+
+				if(com.dotmarketing.portlets.contentlet.business.Contentlet.class.equals(clazz)){
+					Logger.debug(this, "Processing contentlets. This will take a little bit longer...");
+				}
+
+				_xstream = new XStream(new DomDriver());
+
+				//http://jira.dotmarketing.net/browse/DOTCMS-6059
+				if(clazz.equals(DashboardSummary404.class) || clazz.equals(DashboardUserPreferences.class)){
+					_xstream.addDefaultImplementation(com.dotcms.repackage.net.sf.hibernate.collection.Set.class, java.util.Set.class);
+					_xstream.addDefaultImplementation(com.dotcms.repackage.net.sf.hibernate.collection.List.class, java.util.List.class);
+					_xstream.addDefaultImplementation(com.dotcms.repackage.net.sf.hibernate.collection.Map.class, java.util.Map.class);
+					Mapper mapper = _xstream.getMapper();
+					_xstream.registerConverter(new HibernateCollectionConverter(mapper));
+					_xstream.registerConverter(new HibernateMapConverter(mapper));
+				}
+
+				int i= 0;
+				int step = 1000;

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 1000 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)